### ISSUE TITLE
Make logger in `ErrorHandler` optional

### ISSUE THREAD

**Issue by vjik** [yiisoft/error-handler#121]

**Comment by xepozz** [comment]
Why?

**Comment by vjik** [comment]
`ErrorHandler` can work without logger.

**Comment by vjik** [comment]
> There're no reasons to make it optional

Optional always is better than a mandatory one. Creating of object is simplier.

**Comment by vjik** [comment]
> NullLogger is there for this purpose

Yes, optional: `private readonly LoggerInterface $logger = new NullLogger()`
